.github/workflows/security-scanning.yaml

```diff
@@ -0,0 +1,71 @@
+name: Security Scanning
+
+on:
+  push:
+    branches: [ master ]
+    tags: [ '*' ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  snyk:
+    runs-on: ubuntu-latest
+    environment:
+      name: dev
+      url: https://github.com
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@master
+
+    - name: Set Up Snyk
+      uses: snyk/actions/setup@master
+
+    - name: Run Snyk to check for vulnerabilities
+      continue-on-error: true
+      uses: snyk/actions/golang@master
+      env:
+        SNYK_TOKEN: ${{ secrets.SNYK_TOKEN }}
+      with:
+        json: true
+        args: --sarif-file-output=snyk.json
```

Comment:
I think using an absolute path here (perhaps `/tmp/snyk.json`) or then `sarif_file: /github/workspace/snyk.json` later might work around the `upload-sarif` failures.

But maybe something like this would work and be easier to read, too:

```
...

env:
    SARIF_FILE_PATH: /tmp/snyk.json

jobs:
...

    - name: Run Snyk to check for vulnerabilities
      continue-on-error: true
      uses: snyk/actions/golang@master
      env:
        SNYK_TOKEN: ${{ secrets.SNYK_TOKEN }}
      with:
        json: true
        args: --sarif-file-output=${{ env.SARIF_FILE_PATH }}

    - name: Upload result to GitHub Code Scanning
      uses: github/codeql-action/upload-sarif@v2
      with:
       sarif_file: ${{ env.SARIF_FILE_PATH }}
...

```